components/select/Select.razor.cs

```diff
@@ -476,24 +476,6 @@ private void EvaluateDataSourceChange()
                 return;
             }
 
-            // clear DataSource
-            if (DataSource?.Any() == false && SelectOptionItems.Any())
-            {
-                SelectOptionItems.Clear();
-                SelectedOptionItems.Clear();
-
-                Value = default;
-
-                _datasource = DataSource;
-                _dataSourceShallowCopy = new List<TItem>();
-                _dataSourceCopy = new List<TItem>();
-                TypeDefaultExistsAsSelectOption = false;
-
-                OnDataSourceChanged?.Invoke();
-
-                return;
-            }
-
```

Comment:
Why delete this part of the code?

Reply:
Sorry for the delayed response, I will still respond just for documentation purposes...

 I removed this because when a Datasource is used to build the select option list...the "tags" get added to the SelectOptionItems hash set. So when the datasource is just empty, those were getting wiped out but the 2-way binding was not updating so they were still "selected" values. It is a weird edge case but I felt that it was valid for a user to have inputted "tag" options even if the datasource was empty.

Reply:
Thank you for explaining. It makes sense.